### ISSUE TITLE
[BUG]: Can't use RegressionExperiment with the latest version of joblib (1.4.0)

### ISSUE THREAD

**Issue by rfsan** [pycaret/pycaret#3959]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [X] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

Pycaret uses the private function `_format_load_msg` from joblib. This function was removed from joblib version 1.4.0

### Reproducible Example

```python
from pycaret.regression import RegressionExperiment
```

### Expected Behavior

Import works correctly

### Actual Results

```python-traceback
22 from typing import TYPE_CHECKING, Union
     24 from joblib.hashing import Hasher, Pickler
---> 25 from joblib.memory import (
     26     MemorizedFunc,
     27     MemorizedResult,
     28     Memory,
     29     NotMemorizedResult,
     30     _format_load_msg,
     31     filter_args,
     32     format_call,
     33     format_signature,
     34     format_time,
     35     get_func_name,
     36 )
     37 from xxhash import xxh128 as xxh
     39 try:

ImportError: cannot import name '_format_load_msg' from 'joblib.memory'
```

### Installed Versions

<details>
System:
    python: 3.11.8 (main, Feb 28 2024, 07:10:00) [GCC 11.4.0]
   machine: Linux-6.8.0-76060800daily20240311-generic-x86_64-with-glibc2.35

PyCaret required dependencies:
                 pip: 24.0
          setuptools: 69.2.0
             pycaret: 3.3.0
             IPython: 8.23.0
          ipywidgets: 8.1.2
                tqdm: 4.66.2
               numpy: 1.26.4
              pandas: 2.1.4
              jinja2: 3.1.3
               scipy: 1.11.4
              joblib: 1.4.0
             sklearn: 1.4.1.post1
                pyod: 1.1.3
            imblearn: 0.12.2
   category_encoders: 2.6.3
            lightgbm: 4.3.0
               numba: 0.59.1
            requests: 2.31.0
          matplotlib: 3.7.5
          scikitplot: 0.3.7
         yellowbrick: 1.5
              plotly: 5.20.0
    plotly-resampler: Not installed
             kaleido: 0.2.1
           schemdraw: 0.15
         statsmodels: 0.14.1
              sktime: 0.28.0
               tbats: 1.1.3
            pmdarima: 2.0.4
              psutil: 5.9.8
          markupsafe: 2.1.5
             pickle5: Not installed
         cloudpickle: 3.0.0
         deprecation: 2.1.0
              xxhash: 3.4.1
           wurlitzer: 3.0.3

PyCaret optional dependencies:
                shap: Not installed
           interpret: Not installed
                umap: Not installed
     ydata_profiling: Not installed
  explainerdashboard: Not installed
             autoviz: Not installed
           fairlearn: Not installed
          deepchecks: Not installed
             xgboost: Not installed
            catboost: Not installed
              kmodes: Not installed
             mlxtend: Not installed
       statsforecast: Not installed
        tune_sklearn: Not installed
                 ray: Not installed
            hyperopt: Not installed
              optuna: Not installed
               skopt: Not installed
              mlflow: Not installed
              gradio: Not installed
             fastapi: Not installed
             uvicorn: Not installed
              m2cgen: Not installed
           evidently: Not installed
               fugue: Not installed
           streamlit: Not installed
             prophet: Not installed
</details>

**Comment by amotl** [comment]
Dear Rafael,

thanks for reporting. joblib 1.4.0 has been released 7 hours hours ago, and we can confirm the same error.

- https://github.com/crate-workbench/mlflow-cratedb/pull/124#pullrequestreview-1987458192
- https://github.com/crate/cratedb-examples/pull/400#issuecomment-2043682780

With kind regards,
Andreas.

**Comment by amotl** [comment]
Hi again,

after downgrading to Joblib 1.3, we ran into another issue which needed downgrading to Python 3.10 again. Just sharing it for others who are observing similar problems.

- https://github.com/pycaret/pycaret/issues/3960
- https://github.com/crate/cratedb-examples/pull/401

With kind regards,
Andreas.

**Comment by amotl** [comment]
> We ran into another issue which needed downgrading to Python 3.10 again.

In fact, it was an issue with Python 3.11.9 and Dask.
- https://github.com/dask/dask/issues/11038

More details at:
- https://github.com/pycaret/pycaret/issues/3960#issuecomment-2045881422

TLDR; The advised solution is to upgrade to Dask version 2024.4.1.

**Comment by mnpappo** [comment]
Is there any solution to this issue at this moment, I can't run PyCaret on my google colab notebook at all! 

`ImportError                               Traceback (most recent call last)
[<ipython-input-5-a8cb12878b37>](https://localhost:8080/#) in <cell line: 1>()
----> 1 from pycaret.classification import *

6 frames
[/usr/local/lib/python3.10/dist-packages/pycaret/internal/memory.py](https://localhost:8080/#) in <module>
     23 
     24 from joblib.hashing import Hasher, Pickler
---> 25 from joblib.memory import (
     26     MemorizedFunc,
     27     MemorizedResult,

ImportError: cannot import name '_format_load_msg' from 'joblib.memory' (/usr/local/lib/python3.10/dist-packages/joblib/memory.py)`

**Comment by christiantutiven** [comment]
I have the same problem 
"---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
[<ipython-input-23-3be2b1c15beb>](https://localhost:8080/#) in <cell line: 2>()
      1 # import AnomalyExperiment and init the class
----> 2 from pycaret.anomaly import AnomalyExperiment
      3 exp = AnomalyExperiment()

5 frames
[/usr/local/lib/python3.10/dist-packages/pycaret/internal/memory.py](https://localhost:8080/#) in <module>
     23 
     24 from joblib.hashing import Hasher, Pickler
---> 25 from joblib.memory import (
     26     MemorizedFunc,
     27     MemorizedResult,

ImportError: cannot import name '_format_load_msg' from 'joblib.memory' (/usr/local/lib/python3.10/dist-packages/joblib/memory.py)"

**Comment by Julio-CesarAS** [comment]
Hi Guys, i have beem the problem like this, executing in VS code:

-----------------------x-------------------------x---------------------------------

ImportError   Traceback (most recent call last)
Cell In[4], [line 10](vscode-notebook-cell:?execution_count=4&line=10)
      [8](vscode-notebook-cell:?execution_count=4&line=8) from mlflow.tracking import MlflowClient
      [9](vscode-notebook-cell:?execution_count=4&line=9) #from pycaret.classification import setup, create_model, predict_model
---> [10](vscode-notebook-cell:?execution_count=4&line=10) import pycaret.classification as pc
     [12](vscode-notebook-cell:?execution_count=4&line=12) from joblib import Memory

File [c:\Users\julio\anaconda3\envs\infnet-mle2\Lib\site-packages\pycaret\classification\__init__.py:1](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:1)
----> [1](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:1) from pycaret.classification.functional import (
      [2](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:2)     add_metric,
      [3](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:3)     automl,
      [4](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:4)     blend_models,
      [5](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:5)     calibrate_model,
      [6](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:6)     check_drift,
      [7](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:7)     check_fairness,
      [8](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:8)     compare_models,
      [9](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:9)     convert_model,
     [10](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:10)     create_api,
     [11](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:11)     create_app,
     [12](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:12)     create_docker,
     [13](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:13)     create_model,
     [14](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:14)     dashboard,
     [15](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:15)     deploy_model,
     [16](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/classification/__init__.py:16)     ensemble_model,
...
     [36](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/internal/memory.py:36) )
     [37](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/internal/memory.py:37) from xxhash import xxh128 as xxh
     [39](file:///C:/Users/julio/anaconda3/envs/infnet-mle2/Lib/site-packages/pycaret/internal/memory.py:39) try:

ImportError: cannot import name '_format_load_msg' from 'joblib.memory' (c:\Users\julio\anaconda3\envs\infnet-mle2\Lib\site-packages\joblib\memory.py)

----------------------------------x-----------------------------------x-

And the answer is always the same, that this error occurs due to a conflict between the pycaret and joblib libraries. The _format_load_msg function, required by pycaret, has been removed from joblib versions above 1.4.

**Comment by Lawlantosin** [comment]
Hi guys, I have issues importing all the libraries in pycaret.classification on google colab

-------------------------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
[<ipython-input-4-a8cb12878b37>](https://localhost:8080/#) in <cell line: 1>()
----> 1 from pycaret.classification import *

6 frames
[/usr/local/lib/python3.10/dist-packages/pycaret/internal/memory.py](https://localhost:8080/#) in <module>
     23 
     24 from joblib.hashing import Hasher, Pickler
---> 25 from joblib.memory import (
     26     MemorizedFunc,
     27     MemorizedResult,

ImportError: cannot import name '_format_load_msg' from 'joblib.memory' (/usr/local/lib/python3.10/dist-packages/joblib/memory.py)

**Comment by 106firestarter** [comment]
Same here, using pycaret.classification!

**Comment by gchavez83** [comment]
Same here, using from pycaret.regression import *
cannot import name '_format_load_msg' from 'joblib.memory'
pycaret==3.3.0
joblib ==1.4.0

**Comment by MohammedAlmulla** [comment]
Same problem:

ImportError: cannot import name '_format_load_msg' from 'joblib.memory'

PyCaret Version: 3.3.0
joblib version: 1.4.0

**Comment by amotl** [comment]
GH-3963 includes a corresponding fix, also for GH-3960.

In order to use that instantly, if it's important to you, you may want to use this pip package specification to install the package in an ad hoc manner, including the fixes for both dependency flaws. It also works within package metadata specifications like `setup.py`, `setup.cfg`, `pyproject.toml`, or `requirements.txt` files, with recent versions of pip.

```console
pip install --upgrade 'pycaret[parallel] @ git+https://github.com/crate-workbench/pycaret@fix-dependencies-joblib-dask'
```

**Comment by abeoma** [comment]
> Is there any solution to this issue at this moment, I can't run PyCaret on my google colab notebook at all!
> 
> `ImportError Traceback (most recent call last) in <cell line: 1>() ----> 1 from pycaret.classification import *
> 
> 6 frames [/usr/local/lib/python3.10/dist-packages/pycaret/internal/memory.py](https://localhost:8080/#) in 23 24 from joblib.hashing import Hasher, Pickler ---> 25 from joblib.memory import ( 26 MemorizedFunc, 27 MemorizedResult,
> 
> ImportError: cannot import name '_format_load_msg' from 'joblib.memory' (/usr/local/lib/python3.10/dist-packages/joblib/memory.py)`

It might already be resolved, but the workaround on google colab is to install joblib 1.3.0:
```
!pip install joblib==1.3.0 <--
!pip install pycaret==3.3.0
```

**Comment by amotl** [comment]
@abeoma: You can also use Joblib version 1.3.2, so a valid single-line `pip` command can also be:
```
pip install --upgrade 'joblib<1.4' pycaret
```

**Comment by souzapaulino** [comment]
> pip install --upgrade 'joblib<1.4' pycaret

This solved my problem. 
Tks!

**Comment by amotl** [comment]
Please note that GH-3965 is not a viable solution, GH-3967 should work better.

- GH-3967

**Comment by rudro12356** [comment]
> @mnpappo It might already be resolved, but the workaround on google colab is to install joblib 1.3.0:
> 
> ```
> !pip install joblib==1.3.0
> !pip install pycaret==3.3.0
> ```

Thank you. This worked for me. I was trying clustering using Pycaret.

**Comment by amotl** [comment]
PyCaret 3.3.1 has been released. Thank you very much.

-- https://github.com/pycaret/pycaret/releases/tag/3.3.1

**Comment by vasu2497** [comment]
I had run into this issue, installing joblib 1.3.0 and pycaret 3.3.0 resolved the issue. Thanks you
